run-build-functions.sh

```diff
@@ -102,6 +105,34 @@ run_yarn() {
   fi
 
 
+  if [ "$NETLIFY_YARN_WORKSPACES" = "true" ]
+  then
+    echo "NETLIFY_YARN_WORKSPACES feature flag set"
+    # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
+    # to using the global one (which, for now, is alwasy yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
+    if YARN_IGNORE_PATH=1 yarn workspaces info
+    then
+      local package_locations
+      # Extract all the packages and respective locations. .data will be a JSON object like
+      # {
+      #   "my-package-1": {
+      #     "location": "packages/blog-1",
+      #     "workspaceDependencies": [],
+      #     "mismatchedWorkspaceDependencies": []
+      #   },
+      #   (...)
+      # }
+      # We need to cache all the node_module dirs, or we'll always be installing them on each run
+      mapfile -t package_locations <<< "$(YARN_IGNORE_PATH=1 yarn --json workspaces info | jq -r '.data | fromjson | to_entries | .[].value.location')"
```

Comment:
`YARN_IGNORE_PATH=1 yarn workspaces info` is called in the test above. 
Would it make sense to make the test use `--json` and keep the output, to avoid calling `yarn workspaces info` twice? As a tiny performance optimization.

Reply:
Addressed via https://github.com/netlify/build-image/pull/526/commits/073a3db99dd3523a392c1966071442f0f0bf1cdc 👍

Reply:
Now, actually fixed via https://github.com/netlify/build-image/pull/526/commits/8ab88aec42aabe2ec9ecb9681fc0c0a699189d87 😅

Btw, I'm keeping the exit code in a local variable as a way to avoid hitting - https://github.com/koalaman/shellcheck/wiki/SC2181 - let me know what you think @ehmicky

Reply:
Sounds good!